pkg/topology/builder.go

```diff
@@ -745,3 +779,16 @@ func mapCopy(m map[Key]struct{}) map[Key]struct{} {
 
 	return cpy
 }
+
+func removeTrafficSplitsLoop(loopDetected map[*Service][]Key) {
```

Comment:
Still confusing to me that we take a map called loopDetected that doesn't just contain booleans but information that is more complex. I'd suggest renaming this variable, but I don't know how. I'll LGTM though as not to block the PR, but I think this should be changed (or explained)

Reply:
https://github.com/containous/maesh/issues/586